packages/react/src/drafts/TabPanels/TabPanels.tsx

```diff
@@ -0,0 +1,160 @@
+  & > :not([role='tabpanel']) {
+    display: inline-block;
+  }
+
+  &::part(tablist-wrapper) {
+    margin-top: 0;
+    margin-bottom: 16px;
+    border-bottom: 1px solid ${get('colors.border.default')};
+  }
+
+  &:not(:defined) [role='tabpanel'] {
+    margin-top: 17px;
+    display: none;
+  }
+
+  &:not(:defined) [role='tab']:nth-of-type(1)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(1),
+  &:not(:defined) [role='tab']:nth-of-type(2)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(2),
+  &:not(:defined) [role='tab']:nth-of-type(3)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(3),
+  &:not(:defined) [role='tab']:nth-of-type(4)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(4),
+  &:not(:defined) [role='tab']:nth-of-type(5)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(5),
+  &:not(:defined) [role='tab']:nth-of-type(6)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(6),
+  &:not(:defined) [role='tab']:nth-of-type(7)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(7),
+  &:not(:defined) [role='tab']:nth-of-type(8)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(8),
+  &:not(:defined) [role='tab']:nth-of-type(9)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(9),
+  &:not(:defined) [role='tab']:nth-of-type(10)[aria-selected='true'] ~ [role='tabpanel']:nth-of-type(10) {
+    display: block;
+  }
+
+  &:not(:defined):not(:has([aria-selected='true'])) [role='tabpanel']:first-of-type {
+    display: block;
+  }
+
+  &:not(:has([aria-selected='true'])) [role='tab'] ~ [role='tab'] {
+    color: ${get('colors.fg.muted')};
+    background-color: transparent;
+    border: 1px solid transparent;
+  }
+
+  &:not(:has([aria-selected='true'])) [role='tab'],
+  & [role='tab'][aria-selected='true'] {
+    color: ${get('colors.fg.default')};
+    border-color: ${get('colors.border.default')};
+    border-top-right-radius: ${get('radii.2')};
+    border-top-left-radius: ${get('radii.2')};
+    background-color: ${get('colors.canvas.default')};
+  }
+
+  &:not(:defined):not(:has([aria-selected='true'])) [role='tab'] ~ [role='tab'],
+  &:not(:defined):has([aria-selected='true']) [role='tab']:not([aria-selected='true']) {
+    padding: 8px 16px;
+    border-bottom: 1px solid ${get('colors.border.default')};
+  }
+
+  &:not(:defined) :not([role='tabpanel']) {
+    vertical-align: top;
+  }
+
+  ${sx};
+`
+
+export type TabPanelsProps = ComponentProps<typeof TabContainer>
+
+function TabPanels({children, id, ...props}: TabPanelsProps) {
+  // Loop through the chidren, if it's a tab, then add id="{id}-tab-{index}"
+  // If it's a panel, then add aria-labelledby="{id}-tab-{index}"
+  let tabIndex = 0
+  let panelIndex = 0
+
+  const childrenWithProps = React.Children.map(children, child => {
+    if (React.isValidElement<TabPanelsTabProps>(child) && child.type === Tab && !child.props.id) {
+      if (props.selectedTabIndex === tabIndex) {
```

Comment:
Because the selection logic is inside the conditional with `!child.props.id` it appears we won't assign `selected: true` if the `id` prop is set on the `Tab` (ie, if the consumer is using a custom ID here for some reason). It looks like this would break the tab activation behavior?

Reply:
Yeah, I've been wondering about this. Really we don't *want* consumers to set the child ids 🤔 How would you feel about just overriding the id whether the user sets it or not?

I'd have to document it and add how the ids are generated so users could reason about what the ids would be.

Reply:
Hmmmm, I feel like if we have an `id` prop in the type definition, we have to respect it. However, we could 'lie' about / hide the props by excluding `id` from the prop types for the `Tab` and `Panel` components. It would still get assigned when cloned because the underlying styled component would still handle it.